wallet-test/src/test/scala/org/bitcoins/wallet/dlc/DLCExecutionTest.scala

```diff
@@ -0,0 +1,326 @@
+      val offer = offerOpt.get
+
+      assert(tx.inputs.size == 1)
+      assert(tx.outputs.size == 1)
+      assert(tx.outputs.head.scriptPubKey == offer.finalAddress.scriptPubKey)
+      assert(ScriptInterpreter.checkTransaction(tx))
+
+      val sigComponent =
+        WitnessTxSigComponent(tx.asInstanceOf[WitnessTransaction],
+                              UInt32.zero,
+                              fundingTx.outputs.head,
+                              flags)
+      assert(
+        ScriptInterpreter.runVerify(PreExecutionScriptProgram(sigComponent)))
+    }
+  }
+
+  it must "do a dlc mutual close where the recipient wins" in { wallets =>
+    val dlcA = wallets._1.wallet
+    val dlcB = wallets._2.wallet
+
+    for {
+      acceptOpt <- dlcB.dlcAcceptDAO.findByEventId(eventId)
+
+      fundingTx <- dlcA.getDLCFundingTx(eventId)
+
+      closeSig <- dlcB.initDLCMutualClose(eventId, loseSig)
+      tx <- dlcA.acceptDLCMutualClose(closeSig)
+    } yield {
+      assert(acceptOpt.isDefined)
+      val accept = acceptOpt.get
+
+      assert(tx.inputs.size == 1)
+      assert(tx.outputs.size == 1)
+      assert(tx.outputs.head.scriptPubKey == accept.finalAddress.scriptPubKey)
+      assert(ScriptInterpreter.checkTransaction(tx))
+
+      val sigComponent =
+        WitnessTxSigComponent(tx.asInstanceOf[WitnessTransaction],
+                              UInt32.zero,
+                              fundingTx.outputs.head,
+                              flags)
+      assert(
+        ScriptInterpreter.runVerify(PreExecutionScriptProgram(sigComponent)))
+    }
+  }
+
+  it must "do a unilateral close as the initiator" in { wallets =>
+    val dlcA = wallets._1.wallet
+
+    for {
+      (cet, closeTxOpt) <- dlcA.executeDLCForceClose(eventId, winSig)
+
+      fundingTx <- dlcA.getDLCFundingTx(eventId)
+
+    } yield {
+      assert(closeTxOpt.isDefined)
+      val closeTx = closeTxOpt.get
+
+      assert(cet.inputs.size == 1)
+      assert(cet.outputs.size == 1)
+      assert(ScriptInterpreter.checkTransaction(cet))
+      assert(ScriptInterpreter.checkTransaction(closeTx))
+
+      val cetSigComponent =
+        WitnessTxSigComponent(cet.asInstanceOf[WitnessTransaction],
+                              UInt32.zero,
+                              fundingTx.outputs.head,
+                              flags)
+      assert(
+        ScriptInterpreter.runVerify(PreExecutionScriptProgram(cetSigComponent)))
+
+      val closeSigComponent =
+        WitnessTxSigComponent(closeTx.asInstanceOf[WitnessTransaction],
+                              UInt32.zero,
+                              cet.outputs.head,
+                              flags)
+      assert(
+        ScriptInterpreter.runVerify(
+          PreExecutionScriptProgram(closeSigComponent)))
+    }
+  }
+
+  it must "do a unilateral close as the recipient" in { wallets =>
+    val dlcB = wallets._2.wallet
+
+    for {
+      (cet, closeTxOpt) <- dlcB.executeDLCForceClose(eventId, loseSig)
```

Comment:
These tests are all almost the same, it would be nice to have the pattern factored out and turn these tests into two or three liners